### PR TITLE
Add support for Passthrough via USB with baud changes propagating to other port

### PR DESCRIPTION
**Pull request by bugobliterator**
This allows for doing firmware updates to F9P via passthrough among other potential use cases.

### CODE REVIEW

File: libraries/AP_HAL_ChibiOS/Util.h [hunk 1]
```diff
@@ -110,6 +110,7 @@ class ChibiOS::Util : public AP_HAL::Util {
       get system clock in UTC microseconds
      */
     uint64_t get_hw_rtc() const override;
+
```
Comment (bugobliterator) [inline libraries/AP_HAL_ChibiOS/Util.h:113]:
```suggestion

```

File: libraries/AP_HAL_ChibiOS/hwdef/common/usbcfg.c [hunk 2]
```diff
@@ -39,6 +39,11 @@
 /* Virtual serial port over USB.*/
 SerialUSBDriver SDU1;
 
+cdc_linecoding_t linecoding = {
```
Comment (tridge) [inline libraries/AP_HAL_ChibiOS/hwdef/common/usbcfg.c:42]: should be static

File: libraries/AP_HAL_ChibiOS/hwdef/common/usbcfg_dualcdc.c [hunk 3]
```diff
@@ -41,6 +41,16 @@
 SerialUSBDriver SDU1;
 SerialUSBDriver SDU2;
 
+cdc_linecoding_t linecoding = {
```
Comment (tridge) [inline libraries/AP_HAL_ChibiOS/hwdef/common/usbcfg_dualcdc.c:44]: can be static

File: libraries/AP_HAL_ChibiOS/hwdef/common/usbcfg_dualcdc.c [hunk 4]
```diff
@@ -41,6 +41,16 @@
...
+  {0x00, 0x96, 0x00, 0x00},             /* 38400.                           */
+  LC_STOP_1, LC_PARITY_NONE, 8
+};
+
+cdc_linecoding_t linecoding2 = {
```
Comment (tridge) [inline libraries/AP_HAL_ChibiOS/hwdef/common/usbcfg_dualcdc.c:49]: static

Comment (tridge) [reply libraries/AP_HAL_ChibiOS/hwdef/common/usbcfg_dualcdc.c:49]: should be an array of 2 
linecoding[2]

File: libraries/AP_HAL_ChibiOS/hwdef/common/usbcfg_dualcdc.c [hunk 5]
```diff
@@ -446,12 +469,40 @@ static void usb_event(USBDriver *usbp, usbevent_t event) {
...
  */
 static bool requests_hook(USBDriver *usbp) {
-
   if (((usbp->setup[0] & USB_RTYPE_RECIPIENT_MASK) == USB_RTYPE_RECIPIENT_INTERFACE) &&
       (usbp->setup[1] == USB_REQ_SET_INTERFACE)) {
     usbSetupTransfer(usbp, NULL, 0, NULL);
     return true;
   }
+  // process only index=2 commands here - ie the second interface - the first interface is handled via the default handler
+  if ((usbp->setup[0] & USB_RTYPE_TYPE_MASK) == USB_RTYPE_TYPE_CLASS && usbp->setup[4] == 0x02 && usbp->setup[5] == 0x00) {
```
Comment (tridge) [inline libraries/AP_HAL_ChibiOS/hwdef/common/usbcfg_dualcdc.c:478]: can remove the dup code if we make it an array

Comment (tridge) [reply libraries/AP_HAL_ChibiOS/hwdef/common/usbcfg_dualcdc.c:478]: const uint8_t key = usbp->setup[4]/2;

**Review by tridge — CHANGES_REQUESTED** [review]

**Review by tridge — APPROVED** [review]

**Comment by rmackay9** [comment]
I struggled trying to add this to the 4.2 release notes because the description is so short.

**Pull request merged**